lib/chart/animated_line_chart.dart

```diff
@@ -19,22 +19,103 @@ typedef TapText = String Function(String prefix, double y, String unit);
 enum MaxMin { MAX, MIN }
 
 class AnimatedLineChart extends StatefulWidget {
+  /// [LineChart] is in charge of defining the lines which will be drawn in the graph.
   final LineChart chart;
+
+  /// The text being displayed in the tooltip.
+  ///
+  /// Defaults to ```'$x_value: ${y_value.toStringAsFixed(1)} $unit'```, but can be customized.
+  ///
+  /// Example:
+  /// ```dart
+  /// AnimatedLineChart(
+  /// lineChart,
+  /// tapText: (prefix, value, unit) => '$prefix: $value $unit'
+  /// ```
   final TapText? tapText;
+
+  /// Style of tooltip text
   final TextStyle? textStyle;
+
+  /// Background color of tooltip
   final Color toolTipColor;
+
+  /// Color of grid
   final Color gridColor;
+
+  /// List of legends.
+  ///
+  /// If left empty or null, no lengends will be shown at bottom of chart.
+  ///
+  /// The order of defined legends should be the same order of defined lines in [LineChart] to match index,
   final List<Legend>? legends;
+
+  /// Whether markerlines should be shown. Default to 'false'.
   final bool? showMarkerLines;
+
+  /// A list of vertical lines can be defined.
+  /// ```dart
+  /// AnimatedLineChart(
+  /// lineChart,
+  /// verticalMarker: [DateTime.parse('2012-02-27 13:08:00')]
+  /// ```
+  ///
+  /// It is possible to define a maximum of two vertical markerlines.
   final List<DateTime> verticalMarker;
+
+  /// The color of the vertical markerline.
   final Color? verticalMarkerColor;
+
+  /// Icons can be defined, which will be drawn on the vertical markerline if 'verticalMarker' is defined.
+  ///
+  /// The lenght of 'verticalMarkerIcon' must be equal to the length of 'verticalMarker'.
   final List<Icon>? verticalMarkerIcon;
+
+  /// The background color of the icons defined for 'verticalMarkerIcon'. This can be used for unfilled icons.
   final Color? iconBackgroundColor;
+
+  /// Whether shaded areas between the defined markerlines should be shown or not. If true, he shaded area will be in the same color as the markerline.
```

Comment:
Nitpick : If true, he shaded area will be in the same color as the markerline.
```suggestion
  /// Whether shaded areas between the defined markerlines should be shown or not. If true, the shaded area will have the same color as the markerline.
```